Automattic/jetpack
# Fix video block aspect-ratio in frontend

Fixes #17465

#### Changes proposed in this Pull Request:

* Provide the same set of classNames set in the editor to the save function so that block behaves the same in the frontend

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Apply D51151-code
* Go to your sandbox
* Edit a page / post
* Add a 30/30/30 columns block
* Add a video to 1 column
* **Save the post and visit the frontend**
* Video should get full width of the current column
* Height should be adjusted according to the aspect ratio of the original video

Repeat the above for the following themes:
- Varia
- Altofocus
- TwentyTwenty

Especially Altofocus and TwentyTwenty (as also other themes supporting `jetpack-responsive-videos`) handle the embeds height / width with javascript, thus resulting:
1) overriding the css styles
2) better adjusting cause we do not rely on just a bunch of set aspect ratios 
https://github.com/Automattic/jetpack/blob/324cd99629ef2e57a84f1a974085b892004a3434/extensions/blocks/videopress/constants.js#L1-L11

Before | After
-------|------
![](https://cln.sh/BXzpVz+) | ![](https://cln.sh/xublDM+)

#### Proposed changelog entry for your changes:

* Fixes the video block aspect ratio in the frontend (for themes not supporting it)
